net_utils.go

```diff
@@ -207,6 +207,41 @@ func incrementIP(ip net.IP) (result net.IP) {
 	return
 }
 
+// decrementIP returns the given IP 1 1
+func decrementIP(ip net.IP) (result net.IP) {
+	// copy the ip into the result
+	result = make([]byte, len(ip))
+	copy([]byte(result), []byte(ip))
+
+	// Can't decrement an nil or invalid ip
+	if len(result) != 4 && len(result) != 16 {
+		panic(fmt.Errorf("Bad number of bytes for address: %d", len(result)))
```

Comment:
I'm not sure I'm comfortable with a panic here. incrementIP doesn't have the same thing.